### ISSUE TITLE
Exception raised: 'vertices' must be a 2D list or array with shape Nx2

### ISSUE THREAD

**Issue by mgull19** [andycasey/smhr#291]
An exception was encountered in [this session](https://transfer.sh/dGIWM/session.smh) using version e94df1f* on Python 2.7.15 |Anaconda, Inc.| (default, May  1 2018, 18:37:05) [GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]:

````python
Traceback (most recent call last):

  File "/Users/maudegull/Desktop/UROP_FREBEL/smhr/smh/gui/rv.py", line 779, in correct_radial_velocity
    self.parent.normalization_tab.update_rv_applied()

  File "/Users/maudegull/Desktop/UROP_FREBEL/smhr/smh/gui/normalization.py", line 732, in update_rv_applied
    self.draw_order(refresh=False)

  File "/Users/maudegull/Desktop/UROP_FREBEL/smhr/smh/gui/normalization.py", line 1091, in draw_order
    self.norm_plot.reset_zoom_limits()

  File "/Users/maudegull/Desktop/UROP_FREBEL/smhr/smh/gui/mpl.py", line 263, in reset_zoom_limits
    self.draw()

  File "/anaconda2/lib/python2.7/site-packages/matplotlib/backends/backend_agg.py", line 433, in draw
    self.figure.draw(self.renderer)

  File "/anaconda2/lib/python2.7/site-packages/matplotlib/artist.py", line 55, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)

  File "/anaconda2/lib/python2.7/site-packages/matplotlib/figure.py", line 1475, in draw
    renderer, self, artists, self.suppressComposite)

  File "/anaconda2/lib/python2.7/site-packages/matplotlib/image.py", line 141, in _draw_list_compositing_images
    a.draw(renderer)

  File "/anaconda2/lib/python2.7/site-packages/matplotlib/artist.py", line 55, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)

  File "/anaconda2/lib/python2.7/site-packages/matplotlib/axes/_base.py", line 2607, in draw
    mimage._draw_list_compositing_images(renderer, self, artists)

  File "/anaconda2/lib/python2.7/site-packages/matplotlib/image.py", line 141, in _draw_list_compositing_images
    a.draw(renderer)

  File "/anaconda2/lib/python2.7/site-packages/matplotlib/artist.py", line 55, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)

  File "/anaconda2/lib/python2.7/site-packages/matplotlib/collections.py", line 911, in draw
    Collection.draw(self, renderer)

  File "/anaconda2/lib/python2.7/site-packages/matplotlib/artist.py", line 55, in draw_wrapper
    return draw(artist, renderer, *args, **kwargs)

  File "/anaconda2/lib/python2.7/site-packages/matplotlib/collections.py", line 329, in draw
    mpath.Path(offsets), transOffset, tuple(facecolors[0]))

  File "/anaconda2/lib/python2.7/site-packages/matplotlib/path.py", line 135, in __init__
    "'vertices' must be a 2D list or array with shape Nx2")

ValueError: 'vertices' must be a 2D list or array with shape Nx2
````

Below is a screenshot from the application at the time that the exception occurred:

![screenshot](https://transfer.sh/g7vEF/screenshot.png)

**Additional details on how this exception occurred**:

I think it might work but i'm unsure, happened when i tried to cross correlate

**Comment by alexji** [comment]
I think this is not an SMH problem but actually a problem with the star
itself and/or the template you are using to cross-correlate. When I look at
this, I cannot find any feature match between the template and the star (no
H-alpha line anywhere), so something is really wrong.

The error thrown is something about how there is no data to plot, so you
have to fix the cross-correlation first.

**Comment by mgull19** [comment]
uhm it finds the correction and it cross correlates but i can't normalize and that's an issue for several stars in the sample

**Comment by alexji** [comment]
But is the template matching the star in the cross-correlation? I could not
get it to do anything reasonable on the one in the issue, and I couldn't
even find the H-alpha line in that star. I think there is a data reduction
issue with the spectrum, at least for that star.

The error is a plotting thing saying the spectrum data is invalid.

If this is not MIKE data, it is possible that there is a file format error
(e.g. maybe the wavelength unit is nanometers instead of angstroms?)

**Comment by mgull19** [comment]
can i send u the dropbox link of the files and see if you get the same errors?

**Comment by alexji** [comment]
I won't be able to look at it until the weekend but sure

**Comment by alexji** [comment]
This has turned out to be a matplotlib v2.0 error #264

**Comment by alexji** [comment]
This is fixed in #306 (which is still not in the main branch, but closing this issue because it will be there for a while)
